### ISSUE TITLE
[module] cdk-pipeline GitHub Enterprise source action support

### ISSUE THREAD

**Issue by stephen-cloud** [aws/aws-cdk#10461]
I would like to be able to define a CDK Pipeline with a source action for GitHub Enterprise.

GitHub Enterprise appears to only be supported in AWS CodeBuild. I can't find a way to integrate a CodeBuild action into a CDK Pipeline.

### Use Case

We're running GitHub Enterprise at work. I'm a huge fan of CDK Pipeline. 

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by rix0rrr** [comment]
If it's not supported by CodePipeline in the first place, we definitely can't make it happen.

**Comment by vibe** [comment]
@stephen-cloud My solution to this was to have a separate stack, with a codebuild setup to pull from github enterprise. I write the artifacts to an S3 bucket, then I use that s3 bucket as the source in the pipeline stack. I share the buckets between stacks using `ssm.StringParameter`

**Comment by EtoHummin** [comment]
https://aws.amazon.com/about-aws/whats-new/2020/09/aws-codepipeline-now-supports-git-clone-for-source-actions/

I believe this is supported by CodePipeline now

**Comment by venmo-stephen** [comment]
This is more than awesome. Thank you so much!

**Comment by Alchieus** [comment]
Are there plans now to support a GithubEnterpriseSourceAction for GHE now that it can be done via the console?

**Comment by ManAnRuck** [comment]
is there now a way to implement GHE with cdk-pipelines or is there some work in progress?

**Comment by b2jrock** [comment]
@rix0rrr - this can probably be unblocked based on the link that EtoHummin posted.

**Comment by maletor** [comment]
Yes, I am interested in this.

**Comment by maletor** [comment]
Actually, this may be possible now by linking the codestar connection by ARN.

**Comment by laxgoalie392** [comment]
@maletor could you elaborate on this? maybe with a link to an example?

**Comment by maletor** [comment]
I created #14355 to track.

I think bug with the way the role is added. Not working yet.

**Comment by piyush1008** [comment]
I want to create a cdk pipelines with GithubEnterprise , is it  possible or not?

**Comment by aemada-aws** [comment]
Duplicate of https://github.com/aws/aws-cdk/issues/10632 which is solved.
